### ISSUE TITLE
The running result

### ISSUE THREAD

**Issue by 2213500360** [cmavro/ReaRev_KGQA#1]
Hello, thank you for your code. I trained the model on Webqsp and CWQ datasets in the way you described. All the parameters are consistent with the parameters you provided. Finally, the results on Websp are F1: 0.7028, H1: 0.7498, and the results on CWQ are H1: 0.5251, which does not reach the results you provided in the paper. I still can't get results after modifying the batchsize and epoch，What is the reason for this？ THX

**Comment by cmavro** [comment]
Hi, thank you for your interest in our work!

- Indeed, the results for Webqsp that you report seem a bit lower. I am uploading the log file for Webqsp, so you can see whether some hyperparameter is different: 
[Webqsp322.log](https://github.com/cmavro/ReaRev_KGQA/files/10096716/Webqsp322.log)

- For the SOTA results for Webqsp and CWQ, we have performed extensive seed parameter search (unfortunately, I have only kept the models). However, you should be able to reach H@1 scores above 75.8 and 52.5 without extensive tuning. I can try to repeat the process when I find time.

Hope this helps!

**Comment by 2213500360** [comment]
Thank you for your reply. I compared the log you provided and found that all the parameters I used were identical, but my running results were worse， I will try to adjust the parameters again. In addition, I found that the SOTA F1 score in your log reached 71.66, but 70.9 were reported in your paper, What is the reason? THX

**Comment by cmavro** [comment]
We reported the other results when we submitted the paper. We might update the scores in a later version.

**Comment by imhgchoi** [comment]
Hi, thank you very much for this wonderful work. Extending the above issue, I have a question regarding which score you reported.

After running the code, we get six values: best h1 F1 & H1, best f1 F1 & H1, and final epoch F1 & H1. 
Which two of the six values did you report for (76.4 & 70.9) in the paper? 
My guess is (best h1 H1 & best f1 F1). Am I right?

Thanks in advance :)

**Comment by cmavro** [comment]
Hi, thank you for your interest in our work!

Right, this (best h1 for H1 & best f1 for F1) is the straightforward thing to do. 
- However, Webqsp contains only 500 dev questions, which may not provide helpful validation scores. For example, in the log file upload here https://github.com/cmavro/ReaRev_KGQA/issues/1#issuecomment-1328084571 , the model of the final epoch performs the best. 
- For CWQ, we observed that validation scores are much more helpful.

**Comment by imhgchoi** [comment]
I see, thank you for your response :)

**Comment by imhgchoi** [comment]
Hi, one last question. 

May I ask you which score of the two you reported in the paper?
Was it the best H1/F1 val score's test score? Or the final H1/F1 test score?

I've been trying out with the former, but I'm having a hard time reaching over 76.0 H1 test score for WebQSP.
I was curious if maybe you didn't apply early stopping for reporting, as I did see some previous works doing this.

Thank you :)

**Comment by cmavro** [comment]
This is also possible (e.g., start early stopping after some warmup epochs>=100) . Unfortunately, I have not kept the corresponding log files to verify it. 
Please feel free to report the best scores you obtain if you retrain ReaRev.

**Comment by imhgchoi** [comment]
I see, thanks once again for your response :)
